fastly/base_fastly_service_v1.go

```diff
@@ -0,0 +1,482 @@
+				_, err := conn.UpdateVersion(&opts)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		for _, a := range serviceDef.GetAttributeHandler() {
+			if a.MustProcess(d, initialVersion) {
+				if err := a.Process(d, latestVersion, conn); err != nil {
+					return err
+				}
+			}
+		}
+
+		// validate version
+		log.Printf("[DEBUG] Validating Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+		valid, msg, err := conn.ValidateVersion(&gofastly.ValidateVersionInput{
+			Service: d.Id(),
+			Version: latestVersion,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error checking validation: %s", err)
+		}
+
+		if !valid {
+			return fmt.Errorf("[ERR] Invalid configuration for Fastly Service (%s): %s", d.Id(), msg)
+		}
+
+		shouldActivate := d.Get("activate").(bool)
+		if shouldActivate {
+			log.Printf("[DEBUG] Activating Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+			_, err = conn.ActivateVersion(&gofastly.ActivateVersionInput{
+				Service: d.Id(),
+				Version: latestVersion,
+			})
+			if err != nil {
+				return fmt.Errorf("[ERR] Error activating version (%d): %s", latestVersion, err)
+			}
+
+			// Only if the version is valid and activated do we set the active_version.
+			// This prevents us from getting stuck in cloning an invalid version
+			d.Set("active_version", latestVersion)
+		} else {
+			log.Printf("[INFO] Skipping activation of Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+			log.Print("[INFO] The Terraform definition is explicitly specified to not activate the changes on Fastly")
+			log.Printf("[INFO] Version (%v) has been pushed and validated", latestVersion)
+			log.Printf("[INFO] Visit https://manage.fastly.com/configure/services/%s/versions/%v and activate it manually", d.Id(), latestVersion)
+		}
+	}
+
+	return resourceServiceRead(d, meta, serviceDef)
+}
+
+// resourceServiceRead provides service resource Read functionality
+func resourceServiceRead(d *schema.ResourceData, meta interface{}, serviceDef ServiceDefinition) error {
+	conn := meta.(*FastlyClient).conn
+
+	// Find the Service. Discard the service because we need the ServiceDetails,
+	// not just a Service record
+	_, err := findService(d.Id(), meta)
+	if err != nil {
+		switch err {
+		case fastlyNoServiceFoundErr:
+			log.Printf("[WARN] %s for ID (%s)", err, d.Id())
+			d.SetId("")
+			return nil
+		default:
+			return err
+		}
+	}
+
+	s, err := conn.GetServiceDetails(&gofastly.GetServiceInput{
+		ID: d.Id(),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", s.Name)
+	d.Set("comment", s.Comment)
+	d.Set("version_comment", s.Version.Comment)
+	d.Set("active_version", s.ActiveVersion.Number)
+
+	// If CreateService succeeds, but initial updates to the Service fail, we'll
+	// have an empty ActiveService version (no version is active, so we can't
+	// query for information on it)
+	if s.ActiveVersion.Number != 0 {
+
+		for _, a := range serviceDef.GetAttributeHandler() {
```

Comment:
Looks like it [parallels `master`](https://github.com/terraform-providers/terraform-provider-fastly/blob/5b471c351ba16454ad3905e4d8f36fce3c29263b/fastly/resource_fastly_service_v1.go#L566-L571) other than settings could possibly be updated in a different order.